### ISSUE TITLE
visual studio static libs gtest c library MD MT issue

### ISSUE THREAD

**Issue by cyrush** [LLNL/blt#268]
I hit some puzzling linking issues when working on a pure c library, built as an object library 

At first the linking errors looked function export + import related - however I knew I did not need to decorate the functions b/c I was using an object library.

After lots of googling, I eventually found a hint about the underlying issue here:

https://github.com/google/googletest/issues/2074

(A fix for this was merged into gtest earlier this year.)

We have this fix already for CMAKE_CXX_ZZZ flags in SetupCompilerOptions.cmake, we need to also alter the CMAKE_C_ZZZ flags.

I tested the following and it solved my issue:

```
if ( COMPILER_FAMILY_IS_MSVC AND NOT BUILD_SHARED_LIBS )
    foreach(flag_var
            CMAKE_C_FLAGS CMAKE_C_FLAGS_DEBUG CMAKE_C_FLAGS_RELEASE
            CMAKE_C_FLAGS_MINSIZEREL CMAKE_C_FLAGS_RELWITHDEBINFO
            CMAKE_CXX_FLAGS CMAKE_CXX_FLAGS_DEBUG CMAKE_CXX_FLAGS_RELEASE
            CMAKE_CXX_FLAGS_MINSIZEREL CMAKE_CXX_FLAGS_RELWITHDEBINFO)
        if(${flag_var} MATCHES "/MD")
            string(REGEX REPLACE "/MD" "/MT" ${flag_var} "${${flag_var}}")
        endif(${flag_var} MATCHES "/MD")
    endforeach(flag_var)
endif()
```

I'll work on a PR when I get some bandwidth

**Comment by cyrush** [comment]
This has already been fixed :-)

**Comment by cyrush** [comment]
For reference, it was fixed in #256.
